### DISCUSSION TITLE
ROI Annotation by Replacing Estimates.A

### DISCUSSION THREAD

**Discussion by charan1319** [flatironinstitute/CaImAn#1242] [Q&A]
Hello,

I'm attempting to mark ROIs through another program and then pass them in to the cnmf object but am having some trouble doing this. I'm using cellpose to identify cells and am trying to create an array with the same structure as Ain from the demo_seeded_CNMF notebook. Is this the right approach? I'd greatly appreciate any guidance on this! 

Thank you!

Edited for more clear explanation of my current approach

**Comment by EricThomson** [comment]
It would be good for us to offer help with this: I have not yet used cellpose, but I would do what you are doing. I'm not sure what formats cellpose returns as outputs, but basically we have it set up so that we should be able to take in a spatial estimate `Ain` from elsewhere, so it should be a matter of constructing an appropriate interface. If you let me know what the outputs are from cell pose I could probably help!

**Reply by charan1319** [reply]
Sorry about the late response! The output of cellpose that I'm working with right now is an array of the same spacial dimensions as the imaging data, where each pixel is assigned an integer value corresponding to a component if it's part of one, and 0 if it's not part of any component. Since Ain takes in a binary matrix of shape (num_pixels, num_components), I'm just one-hot embedding the values from cellpose to match the required shape. Just to confirm for flattening the output of cellpose, Ain expects row-major ordering of pixels right?

**Reply by charan1319** [reply]
Update: I'm getting the following error when running cnm_seeded.fit(images) after following the steps shown in the demo_seeded_CNMF notebook: Exception: You need to provide estimate of C and f. 

Looking through the code for the update_spatial_component and computing_indicator functions it looks like the object is already expected to have C and f, but I don't see where in the example notebook these are set. Could you please point me in the right direction regarding this error? 

Thank you!

**Reply by EricThomson** [reply]
Hi @charan1319 just getting back to things from 2023. I will take a look at this soon (in the coming weeks).

**Reply by ethanbb** [reply]
Probably irrelevant at this point, but I had this open in a tab so in case it helps @charan1319 or anyone else:

The Ain must have dtype bool in order to get to the code path that doesn't expect C and f to also be provided. I'm not sure whether this is intentional.

Also the pixels should be flattened in column-major (F) order, not row-major (C).
